app/README.md

```diff
@@ -52,9 +52,127 @@ committing changes, otherwise the tests will be skipped on CI.
+Guide, explaining all the different variations and use cases a component can be
+used.
+
+
+#### File structure
+
+To be able to find code and make changes quickly we organize our components in
+groups defined by use-case, re-usability and complexity. These groups are:
+
+  - `DesignSystem/Primitives`: simple, yet highly reusable components like
+    typography, buttons, form elements, spacing, positioning and other
+    utilities.
+
+    Components in this group are usually just wrappers around standard HTML
+    elements with custom styling.
+
+    There are currently two ways of organizing primitives:
+
+      - all-in-one components where the type of the component is passed down
+        via a `variant` prop. This is for components which have a very
+        similar markup, but whose styling differs across variants.
+        Examples in this category are: buttons, typography and positioning
+        helpers.
+
+      - namespaced components, where the markup is very different across
+        variants, like: form elements and icons.
+
+    To figure which way to write a new primitive component, start by looking at
+    how it's going to be used in the code and then go for the better
+    ergonomics.
+
+    All public primitives are exported via a central `index.js` file, which
+    makes usage as easy as:
+
+    ```
+    <script>
+      import { Button, Text, Icon, Input } from "../DesignSystem/Primitives";
+    </script>
+
+    <Icon.Home />
+    <Button variant="secondary">OK</Button>
+    <Text variant="hugeTitle">Radicle</Text>
+    ```
+
+
+  - `DesignSystem/Components`: reusable low-to-high complexity components,
+    layouts and single-use fragments which are extracted from screens or other
+    components for the sake of code readability.
+
+    Sub-folders in this category should only be created for breaking up larger
+    components into smaller fragments. If a component is broken up in
+    fragments, make sure to only export the component for public use, not the
+    fragments.
+
+    Single-use components that are extracted from screens and used by only one
+    screen should be prefixed with the name of the screen, for example:
+    `DesignSystemGuideSection.svelte`.
+
+
+There is a third group of components, which is not part of the Design System:
+"Screens". Screens bring together components from the Design System forming
+what a user sees in the UI as a whole screen. More complex UX flows can be
+broken down into multiple sub-screens, in this case a top-level screen should
+contain common data fetching and routing for the sub-screens.
+
+When multiple screens share the same layout, it should be extracted into a
+separate component. Layout components are suffixed with "Layout", like so:
+`DesignSystem/Components/ModalLayout.svelt`.
+
+File and directory name casing:
+  - Svelte components and directories containing components - PascalCase
+  - everything else: `*.js` files and folders - camelCase
+
```

Comment:
Currently there is a blind spot about `.css` files, how they are structured, when and where to make changes and how they play together with the design system. Maybe we can expand on that, too.

Reply:
👍 forgot to add the CSS guidelines, will work on it.

Reply:
:heart: